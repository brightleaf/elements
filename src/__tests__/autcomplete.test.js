import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { toHaveClass } from '@testing-library/jest-dom'
import { AutoComplete } from '../'

expect.extend({ toHaveClass })

describe('AutoComplete component', () => {
  afterEach(cleanup)
  it('should render', () => {
    const items = [
      { name: 'First', id: 1 },
      { name: 'Second', id: 2 },
      { name: 'Third', id: 3 },
    ]
    const { container } = render(
      <AutoComplete
        list={items}
        onSelect={item => {}}
        onValueChange={e => {}}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('autocomplete')
  })
})
