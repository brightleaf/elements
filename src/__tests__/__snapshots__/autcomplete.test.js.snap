// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete component should render 1`] = `
<div
  class="autocomplete control"
>
  <div
    class="control is-clearfix has-icons-right"
  >
    <input
      autocomplete="off"
      class="input"
      data-lpignore="true"
      placeholder="Placeholder Text"
      type="text"
      value=""
    />
    <span
      class="icon is-small is-right"
    >
      <i
        class="fa fa-chevron-down"
      />
    </span>
  </div>
   
  <div
    class="dropdown-menu animated is-hidden"
  >
    <div
      class="dropdown-content"
    />
  </div>
</div>
`;
